File: core/chatwoot/resources/labels.rb
# frozen_string_literal: true

module Chatwoot
  module Resources
    class Labels
      def initialize(client, conversation_id)
        @client = client
        @conversation_id = conversation_id
      end

      def list
        @client.get("/conversations/#{@conversation_id}/labels")
      end

      def add(labels)
        @client.post("/conversations/#{@conversation_id}/labels", { labels: Array(labels) })
      end
    end
  end
end

File: internal/sdk/labels.go
package sdk

import (
	"bytes"
	"encoding/json"
	"fmt"
)

type LabelsService struct {
	client         *Client
	conversationID int
}

type LabelsResponse struct {
	Payload []string `json:"payload"`
}

func (s *LabelsService) List() ([]string, error) {
	path := fmt.Sprintf("/conversations/%d/labels", s.conversationID)
	var resp LabelsResponse
	if err := s.client.Get(path, nil, &resp); err != nil {
		return nil, err
	}
	return resp.Payload, nil
}

type AddLabelsRequest struct {
	Labels []string `json:"labels"`
}

func (s *LabelsService) Add(labels []string) ([]string, error) {
	body := AddLabelsRequest{Labels: labels}

	jsonBody, err := json.Marshal(body)
	if err != nil {
		return nil, err
	}

	path := fmt.Sprintf("/conversations/%d/labels", s.conversationID)
	var resp LabelsResponse
	if err := s.client.Post(path, bytes.NewReader(jsonBody), &resp); err != nil {
		return nil, err
	}

	return resp.Payload, nil
}
